Library/Homebrew/download_strategy.rb

```diff
@@ -397,6 +397,39 @@ def curl(*args, **options)
     args << "--connect-timeout" << "5" unless mirrors.empty?
     super(*args, **options)
   end
+
+  module S3DownloadExtension
+    def _fetch
+      # Put the aws gem requirement here (vs top of file) so it's only
+      # a dependency of S3 users, not all Homebrew users
+      require "rubygems"
+      begin
+        require "aws-sdk-v1"
+      rescue LoadError
+        onoe "Install the aws-sdk gem into the gem repo used by brew."
```

Comment:
Could use `def install_gem_setup_path!` for this (see `utils.rb`)

Reply:
Thanks for the comment. This `_fetch` behavior is completely unchanged (I literally just cut and paste it into the new module).

I'm happy to look into improving it though, thanks!